analytics.py

```diff
@@ -0,0 +1,244 @@
+            max_population = featureEntry["properties"]["pop_max"]
+            city = featureEntry["properties"]["nameascii"]
+
+
+    return city, max_population
+
+def write_your_own(gj):
+    """
+    This function finds the least populated city, pop_min
+    """
+    featureList = gj["features"]
+    minPop = 999999999
+    for featureEntry in featureList:
+        #feature["properties"]["pop_min"] for feature in self.gj["features"]
+        if featureEntry["properties"]["pop_min"] < minPop:
+            minPop = featureEntry["properties"]["pop_min"]
+            city = featureEntry["properties"]["nameascii"]
+  #  minn = min(featureEntry["properties"]["pop_min"])
+#    print(minn)
+    return city, minPop
+
+def mean_center(points):
+    """
+    Given a set of points, compute the mean center
+
+    Parameters
+    ----------
+    points : list
+         A list of points in the form (x,y)
+
+    Returns
+    -------
+    x : float
+        Mean x coordinate
+
+    y : float
+        Mean y coordinate
+    """
+
+    #find the average of all the X points in the list
+
+  #  x_sum = sum(points[0])
+    #points_length = len(points)
+
+    sums = map(sum,zip(*points)) # returns iterable object of type map
+    sumsL = list(sums)
+    avgs = map(lambda xy: xy/len(points),sumsL)
+    avgsL = list(avgs)
+    x = avgsL[0]
+    y = avgsL[1]
+
+    return x,y
+
+def average_nearest_neighbor_distance(points):
+    """
+    Given a set of points, compute the average nearest neighbor.
+
+    Parameters
+    ----------
+    points : list
+             A list of points in the form (x,y)
+
+    Returns
+    -------
+    mean_d : float
+             Average nearest neighbor distance
+
+    References
+    ----------
+    Clark and Evan (1954 Distance to Nearest Neighbor as a
+     Measure of Spatial Relationships in Populations. Ecology. 35(4)
+     p. 445-453.
+    """
+
+    #d_i is the set of all of the distances between i and it's closest neighbor.
+    #then, between all of those distances, you divide by the number of points.
+
+    #find the distance between a point i and every other point
+    shDistL =[]       #list of shortest distances
+    mean_sum = 0
+    for point in points:
+        shortestDistance = 9999999999
+        for dpoint in points:
+            if point != dpoint:
+                dist = euclidean_distance(point, dpoint)
+                if(shortestDistance > dist):
+                    shortestDistance = dist
+        #now add the shortest distance of that point before it moves on to a new point
+        shDistL.append(shortestDistance)
+        mean_sum = shortestDistance + mean_sum
+
+    #once shDistL has all of the shortest distances, find the sum of those and divide by the number of points sums = map(sum,zip(*points))
+    #sums = map(sum,shDistL) #list like [x]
+   # print(type(sums))
+    print(shDistL)
+    sums = sum(shDistL)
+    mean_d = mean_sum/len(shDistL)
```

Comment:
Why are you computing `sums`, but then using mean_sum?  I would remove mean_sum entirely and just take the sum of the list of the shortest distances divided by the total length of that list.